aws/resource_aws_cloudfront_key_group.go

```diff
@@ -0,0 +1,146 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsCloudFrontKeyGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudFrontKeyGroupCreate,
+		Read:   resourceAwsCloudFrontKeyGroupRead,
+		Update: resourceAwsCloudFrontKeyGroupUpdate,
+		Delete: resourceAwsCloudFrontKeyGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"comment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"items": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudFrontKeyGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	input := &cloudfront.CreateKeyGroupInput{
+		KeyGroupConfig: expandCloudFrontKeyGroupConfig(d),
+	}
+
+	log.Println("[DEBUG] Create CloudFront key group:", input)
+
+	output, err := conn.CreateKeyGroup(input)
+	if err != nil {
+		return fmt.Errorf("error creating CloudFront key group: %s", err)
+	}
+
+	d.SetId(aws.StringValue(output.KeyGroup.Id))
+	return resourceAwsCloudFrontKeyGroupRead(d, meta)
+}
+
+func resourceAwsCloudFrontKeyGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+	input := &cloudfront.GetKeyGroupInput{
+		Id: aws.String(d.Id()),
+	}
+
+	output, err := conn.GetKeyGroup(input)
+	if err != nil {
+		if isAWSErr(err, cloudfront.ErrCodeNoSuchResource, "") {
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16796

```suggestion
		if !d.IsNewResource && isAWSErr(err, cloudfront.ErrCodeNoSuchResource, "") {
```